java/client/src/main/java/glide/api/models/commands/StreamTrimOptions.java

```diff
@@ -0,0 +1,155 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+
+/**
+ * Optional arguments for {@link StreamBaseCommands#xadd(String, Map, StreamAddOptions)} and {@link
+ * StreamBaseCommands#xtrim(String, TrimLimit)}
+ *
+ * @see <a href="https://redis.io/commands/xtrim/">redis.io</a>
+ */
+public final class StreamTrimOptions {
+    public static final String TRIM_MAXLEN_REDIS_API = "MAXLEN";
+    public static final String TRIM_MINID_REDIS_API = "MINID";
+    public static final String TRIM_EXACT_REDIS_API = "=";
+    public static final String TRIM_NOT_EXACT_REDIS_API = "~";
+    public static final String TRIM_LIMIT_REDIS_API = "LIMIT";
+
+    public abstract static class TrimLimit {
+
+        /**
+         * If <code>true</code>, the stream will be trimmed exactly. Equivalent to <code>=</code> in the
+         * Redis API. Otherwise, the stream will be trimmed in a near-exact manner, which is more
+         * efficient, equivalent to <code>~</code> in the Redis API.
+         */
+        protected boolean exact;
+
+        /** If set, sets the maximal amount of entries that will be deleted. */
+        protected Long limit;
+
+        protected abstract String getMethod();
+
+        protected abstract String getThreshold();
+
+        protected List<String> getRedisApi() {
+            List<String> optionArgs = new ArrayList<>();
+
+            optionArgs.add(this.getMethod());
+            optionArgs.add(this.exact ? TRIM_EXACT_REDIS_API : TRIM_NOT_EXACT_REDIS_API);
+            optionArgs.add(this.getThreshold());
+
+            if (this.limit != null) {
+                optionArgs.add(TRIM_LIMIT_REDIS_API);
+                optionArgs.add(this.limit.toString());
+            }
+
+            return optionArgs;
+        }
+    }
+
+    /** Option to trim the stream according to minimum ID. */
+    @RequiredArgsConstructor
+    public static class MinId extends TrimLimit {
+        /** Trim the stream according to entry ID. Equivalent to <code>MINID</code> in the Redis API. */
+        private final String threshold;
+
+        /**
+         * Create a trim option to trim stream based on stream ID.
+         *
+         * @param exact Whether to match exactly on the threshold.
+         * @param threshold Comparison id.
+         */
+        public MinId(boolean exact, @NonNull String threshold) {
+            this.threshold = threshold;
+            this.exact = exact;
+        }
+
+        /**
+         * Create a trim option to trim stream based on stream ID.
+         *
+         * @param exact Whether to match exactly on the threshold.
+         * @param threshold Comparison id.
+         * @param limit Max number of stream entries to be trimmed.
+         */
+        public MinId(boolean exact, @NonNull String threshold, long limit) {
+            this.threshold = threshold;
+            this.exact = exact;
+            this.limit = limit;
```

Comment:
Doesn't this mean you can specify "=" and limit at the same time? You should only be allowed to specify limit if using "~".

Reply:
You can.  If it's a request error on the Redis side, that's fine.  But the documentation doesn't indicate that there's a restriction here.

Reply:
Maybe I'll include an example in the IT for this...